stack.go

```diff
@@ -151,3 +151,21 @@ func funcname(name string) string {
 	i = strings.Index(name, ".")
 	return name[i+1:]
 }
+
+// NewStack is for library implementers that want to generate a stack trace.
+// Normally you should use AddStack to get an error with a stack trace.
+// Stack trace information can be generated by calling this function.
+// It can then be turned into a stack trace by calling .StackTrace()
+//
+// This function takes an optional argument for the position in the stack to start from.
+// This avoids putting stack generation function calls in the stack trace.
+// It defaults to 1, which removes the NewStack function from the stack trace.
+func NewStack(position ...int) StackTracer {
```

Comment:
I think variable args should be used for multiple values, not for optional.
How about that we make `position` a fixed argument, then add 1 internally?
The default usage is `NewStack(0)`.

Reply:
I agree its not great, but that's golang :(. I think `NewStack(int)` is possible, but we cannot have any internal addition. In that case I would just consider exposing two functions, one that uses just 1.

Reply:
@coocood Okay, I am requiring the argument now. But there is no internal addition (beyond what was already used in callers()), so normal usage will be `NewStack(1)`.